security/fma-incident-response-improvements.md

```diff
@@ -0,0 +1,254 @@
+
+- **DelayedWETH**: Contract is basically unchanged and the diff is so minor there's really no genuine concern for any sort of failure.
+
+- **AnchorStateRegistry**: Since this change doesn't actually make the ASR a critical dependency yet (except for `isGameRespected`), the only real failure modes are:
+  - The anchor state not being updated correctly (documented in [FM3](#fm3-anchor-state-fails-to-progress-within-a-time-bound))
+  - Incorrect anchor state being set
+  - (We're already planning to add monitoring for anchor state getting too old and we can easily add monitoring for anchor state being invalid)
+
+- **OptimismPortal**: Very minor changes, with two main failure modes:
+  - The new incident response functionality being misused (documented in [FM2](#fm2-legacy-runbookresponse-confusion))
+  - A critical issue that allows a game that isn't respected to be used (either a bug in the changes or a bug in `ASR.isGameRespected`)
+
+Below are references for this project:
+
+- Design Docs
+  - [Anchor State Poisoning Protection](https://github.com/ethereum-optimism/design-docs/blob/b807dc5880f48a3631222e7f519c02a7668f2493/protocol/proofs/anchor-state-poison-protection.md)
+  - [DelayedWETH Expansion](https://github.com/ethereum-optimism/design-docs/blob/b807dc5880f48a3631222e7f519c02a7668f2493/protocol/proofs/delayed-weth-expansion.md)
+  - [FDG Refund Mode](https://github.com/ethereum-optimism/design-docs/blob/b807dc5880f48a3631222e7f519c02a7668f2493/protocol/proofs/fdg-refund-mode.md)
+  - [Withdrawal Invalidation Mitigation](https://github.com/ethereum-optimism/design-docs/blob/b807dc5880f48a3631222e7f519c02a7668f2493/protocol/proofs/withdrawal-invalidation-mitigation.md)
+- Specifications
+  - [Fault Dispute Game/Bond Incentives](https://github.com/ethereum-optimism/specs/pull/560/files)
+  - [Portal](https://github.com/ethereum-optimism/specs/blob/ab35e8181f34614503a03e18a282cce69ea634fe/specs/fault-proof/stage-one/optimism-portal.md)
+  - [AnchorStateRegistry](https://github.com/ethereum-optimism/specs/blob/7706b68c6f8f4172f9396c03175e6c8cb299bfbc/specs/fault-proof/stage-one/anchor-state-registry.md)
+  - [DelayedWETH](https://github.com/ethereum-optimism/specs/blob/a0c94920a3c3b6b3527e4f9c3f07b787cd9fe2ad/specs/fault-proof/stage-one/bond-incentives.md#delayedweth)
+
+## FM1: Contract deployment and/or initialization failures
+
+### Description
+
+Any of the modified contracts could be deployed or initialized incorrectly leading to a variety of failure modes, depending on the exact contract impacted.
+
+### Risk Assessment
+
+- Impact: HIGH/CRITICAL
+  - Reasoning: Highly depends on the exact failure mode but can be critical in the worst case.
+- Likelihood: LOW
+  - Reasoning: Very likely to be covered by both OPCM, testing, and auditing.
+
+### Mitigations
+
+- Validation via OPCM
+- Integration testing
+- 3rd party audit
+- Testnet deployment
```

Comment:
Are there certain OPCM validations, tests, and audits we can link to as evidence these mitigations are implemented? Or should we track implementing as an action item?

Reply:
I don't really love this kind of failure mode, since it basically just says "we did configured OPCM wrong and now the system is configured wrong" - there's no clear place in OPCM where you're really supposed to prevent this from happening. We can point at the checks that we wrote in OPCM but that seems insufficient in general. I feel like this is very related to our other conversation of how we solve the problem of confirming that a deployment is configured correctly.

Reply:
Yea in general FMAs should be for feature-specific failure modes, not generic ones, which are accounted for in https://github.com/ethereum-optimism/design-docs/blob/main/security/fma-generic-contracts.md (which doesn't have OPCM info yet). If there is something unique about these features worth calling out they can get mentioned in this failure mode, otherwise I think we don't really need it since it's very generic

Reply:
Ok let's just remove this since it's generic

Reply:
removed in Added in dd17b25